src/engine/pieces/queen.ts

```diff
@@ -1,13 +1,27 @@
 import Piece from './piece';
 import Player from '../player';
 import Board from '../board';
+import Square from '../square';
 
 export default class Queen extends Piece {
     public constructor(player: Player) {
         super(player);
     }
 
     public getAvailableMoves(board: Board) {
-        return new Array(0);
+        let availableMoves = new Array<Square>();
+        let currentPosition = board.findPiece(this);
+        
+        this.pathCheck(-1,0,availableMoves,currentPosition,board);
+        this.pathCheck(1,0,availableMoves,currentPosition,board);
+        this.pathCheck(0,1,availableMoves,currentPosition,board);
+        this.pathCheck(0,-1,availableMoves,currentPosition,board);
+
+        this.pathCheck(-1,-1,availableMoves,currentPosition,board);
+        this.pathCheck(-1,1,availableMoves,currentPosition,board);
+        this.pathCheck(1,-1,availableMoves,currentPosition,board);
+        this.pathCheck(1,1,availableMoves,currentPosition,board);
+
```

Comment:
Really nice how you can use the same `pathCheck `function

Similarly to what I said [here](https://github.com/emma-bragg/ts-chessington/pull/1/files#r1666692716) you could simplify this

Reply:
would be great to write helper that takes in a set of row/ col directions to check and then runs path check on all of them

then bishop, rook and queen could all use that

Reply:
Fixed